web/html/src/core/channels/api/use-mandatory-channels-api.ts

```diff
@@ -0,0 +1,98 @@
+import { useState } from "react";
+import { JsonResult } from "utils/network";
+import Network from "utils/network";
+import { ChannelType } from "core/channels/type/channels.type";
+import { ChannelsDependencies } from "core/channels/utils/channels-dependencies.utils";
+import {
+  processChannelDependencies,
+  dependenciesTooltip as dependenciesTooltipInternal,
+} from "core/channels/utils/channels-dependencies.utils";
+import { MessageType } from "components/messages";
+
+const msgMap = {
+  base_not_found_or_not_authorized: t("Base channel not found or not authorized."),
+  child_not_found_or_not_authorized: t("Child channel not found or not authorized."),
+  invalid_channel_id: t("Invalid channel id"),
+};
+
+type FetchMandatoryChannelsProps = {
+  base: any;
+  channels: Array<ChannelType>;
+};
+
+export type RequiredChannelsResultType = {
+  requiredChannels: Map<number, Set<number>>;
+  requiredByChannels: Map<number, Set<number>>;
+  dependenciesTooltip: Function;
+};
+
+export type UseMandatoryChannelsApiReturnType = {
+  requiredChannelsResult: RequiredChannelsResultType;
+  isDependencyDataLoaded: boolean;
+  fetchMandatoryChannelsByChannelIds: Function;
+};
+
+const useMandatoryChannelsApi = (): UseMandatoryChannelsApiReturnType => {
+  const [messages, setMessages] = useState([] as MessageType[]);
```

Comment:
Is this different than: `useState<MessageType[]>([])`?
If it's all the same, I think we should have the preference of one or the other

Reply:
The resulting types are the same, I agree that the `useState<T>` pattern is better. 👍